src/main/java/appeng/tile/storage/TileIOPort.java

```diff
@@ -477,13 +511,21 @@ private boolean matches(final FullnessMode fm, final IMEInventory src) {
         }
 
         if (fm == FullnessMode.EMPTY) {
-            return myList.isEmpty();
+            // If filling from network and mode is set to "Move on empty", move when network is empty
+            if (om == OperationMode.FILL) {
+                return didWork;
```

Comment:
Why does work need to be done to move here?

Reply:
It's to differentiate it from the "Move when work done" mode. If it returned `True` instead, it would move the cell even if nothing was moved, which is the behavior you get with the existing mode. Basically, the "Fill from network" & "Move on empty" can be read as "Move to output if network *was emptied* by an IO port action *and it was not empty initially*"